### PR TITLE
OpenFHE e2e test harness

### PR DESCRIPTION
**Pull request by j2kun**
Fixes https://github.com/google/heir/issues/385

### CODE REVIEW

File: tests/openfhe/end_to_end/binops_test.cc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+namespace mlir {
+namespace heir {
+namespace openfhe {
+
+TEST(BinopsTest, TestInput1) {
+  CCParams<CryptoContextBGVRNS> parameters;
+  parameters.SetMultiplicativeDepth(2);
+  parameters.SetPlaintextModulus(65537);
+  CryptoContext<DCRTPoly> cryptoContext = GenCryptoContext(parameters);
```
Comment (j2kun) [inline tests/openfhe/end_to_end/binops_test.cc:19]: This line currently fails with

```
"external/openfhe/src/core/include/math/nbtheory-impl.h:332 
FirstPrime: Requested bit length 60 exceeds maximum allowed length 57"
```

**Comment by j2kun** [comment]
@asraa this should be ready! we might want to chat about it a little bit, because the setup is bazel-first, and has an associated macro/rule.

**Comment by j2kun** [comment]
CC @Maokami

File: tests/openfhe/end_to_end/BUILD [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    default_applicable_licenses = ["@heir//:license"],
+    default_visibility = ["//visibility:public"],
+)
+
+openfhe_end_to_end_test(
+    name = "binops_test",
+    generated_lib_header = "binops_lib.h",
+    mlir_src = "binops.mlir",
+    test_src = "binops_test.cc",
+)
```
Comment (j2kun) [inline tests/openfhe/end_to_end/BUILD:15]: tags=manual/notap

File: .bazelrc [hunk 3]
```diff
@@ -1,5 +1,7 @@
-common --action_env=BAZEL_CXXOPTS=-std=c++17
-common --cxxopt='-std=c++17'
+# gnu++17 is needed to avoid issues with openfhe support for __int128_t
+# see https://github.com/openfheorg/openfhe-development/issues/669
```
Comment (asraa) [inline .bazelrc:2]: could you please file a quick TODO in the WORKSPACE for openfhe to remove this when they update their release?

**Review by asraa — APPROVED** [review]

**Comment by j2kun** [comment]
Fixed in https://github.com/google/heir/commit/805c44755a28cb643916ee827acaea805aab4197

(copybara issues)

**Pull request closed without merging**